haystack/nodes/preprocessor/preprocessor_new.py

```diff
@@ -0,0 +1,481 @@
+        clean_substrings: Optional[List[str]] = None,
+    ):
+        """
+        Performs document cleaning and splitting.
+
+        :param documents: the documents to process.
+
+        :param clean_whitespace: Strip whitespaces before or after each line in the text.
+
+        :param clean_empty_lines: Remove more than two empty lines in the text.
+
+        :param clean_regex: Remove the specified regex matches from the text. For example, `clean_regex='[0-9]'`
+                            removes all digits from the document's content, and `clean_regex='(a string|another string)'`
+                            removes all occurrences of either string from the document content.
+
+        :param clean_header_footer: Use a heuristic to remove footers and headers across different pages by searching
+                                    for the longest common string. This heuristic uses exact matches and
+                                    works well for footers like "Copyright 2019 by The Author", but won't detect "Page 3 of 4"
+                                    or similar. Use 'clean_regex' to detect such headers.
+
+        :param header_footer_n_chars: Headers and footers will only be searched in the first and last n_chars of each page.
+                                    Defaults to 50.
+
+        :param header_footer_pages_to_ignore: Indices of the pages that the header/footer search algorithm should ignore.
+                                            For example, to ignore the first two pages and the last three,
+                                            set `pages_to_ignore=[0, 1, -3, -2, -1]`. By default it ignores no pages.
+
+        :param split_by: Unit for splitting the document. Can be 'character', 'token', 'word', 'sentence', 'paragraph', 'page', 'regex'.
+
+        :param split_regex: If split_by="regex", provide a regex matching the separator. For example if the document
+                            should be split on "--my separator--", this field should be `split_regex="--my separator--"`.
+
+        :param split_length: The maximum number of the above split unit (like word, sentence, page and so on) that are allowed in one document.
+                                For instance, if `split_lenght=10` and `split_by="sentence"`, then each output document will contain 10 sentences.
+
+        :param split_overlap: Units (for example words or sentences) overlap between two adjacent documents after a split.
+                                For example, if `split_by="word" and split_length=5 and split_overlap=2`, then the splits would be like:
+                                `[w1 w2 w3 w4 w5, w4 w5 w6 w7 w8, w7 w8 w10 w11 w12]`.
+                                Set the value to 0 to ensure there is no overlap among the documents after splitting.
+
+        :param split_max_chars: Absolute maximum number of chars allowed in a single document. Reaching this boundary
```

Comment:
It doesn't look like `split_max_chars` is an input to `run_batch`. Is this something that we would like to add?